src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Color.cs

```diff
@@ -1158,6 +1158,15 @@ private void ComputeNativeValues(int numChannels)
         private struct MILColorF // this structure is the "milrendertypes.h" structure and should be identical for performance
```

Comment:
Not marked as `Pack=1`. Does this really get appropriate treatment from `ValueType.Equals()`  ?

Reply:
Isn't pack just for alignment of fields within the struct when you mix fields of different sizes? I don't think the default packing will leave gaps for 4 floats on any platform.

[doc](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.structlayoutattribute.pack) says `Each field must align with fields of its own size (1, 2, 4, 8, etc., bytes) or the alignment of the type, whichever is smaller.` In other words the Pack should only reduce gaps not create them. If all your fields have the same size than Pack is irrelevant as far as I'm aware.

Reply:
There are checks built into ValueType equality that help determine whether to reflect or use byte-comparisons.  The MethodTable contains the flags for all of this I believe.

See https://github.com/dotnet/coreclr/blob/a64cb0a41a4ebeb4a61b8b0f1f2eeeff8dd539c9/src/System.Private.CoreLib/src/System/ValueType.cs#L44

And follow the chain into the implementations of CanCompareBits and FastEqualsCheck.

These are related to how packed the struct is, see https://github.com/dotnet/coreclr/blob/54d9d9b2698c9628f43494e46f855996b31838bd/src/vm/comutilnative.cpp#L2063.

It's slightly hard to see where all of this gets set and the exact criteria.  The MSDN documentation seems to say as long as none of the fields are reference types, you can use the fast comparisons.  But I am not 100% sure that is true from reading the above.

Reply:
I also see this: https://github.com/dotnet/coreclr/blob/0bdf613f09a3bf409717beb939195b39e40c6828/src/vm/class.cpp#L413

And:
https://github.com/dotnet/coreclr/blob/5885379686cd854361735a09907d654e80f44f05/src/vm/class.h#L1946

So I'm not sure if fast comparisons are turned off all the time or not.

Reply:
@ojhad, I approve the PR and this work need to be merge tonight, and since this is about GetHashCod - but I didn't get a good response for this question. As @rladuca points out, the checks for `Pack` are baked into the ValueType equality.... please open an issue to investigate this before merging the PR/closing this comment.

Reply:
Created Issue for follow up:
https://github.com/dotnet/wpf/issues/1338